lib/tasks/cms/cms.rake

```diff
@@ -210,6 +210,11 @@ namespace :cms do
         puts "\t++ Fixing pagination href in #{page.id}/shared/pagination"
       end
     end
+
+    desc 'Fixes empty titles in sections'
+    task :section_empty_titles => :environment do
+      CMS::Section.unscoped.where(title: ['', nil]).update_all("title = id")
```

Comment:
Why did you chose the `id` for filling the missing value? I think the `system_name` would be a more logical choice? Although, maybe I am missing some scenario where it might not work...

Reply:
I'm choosing the id for two reasons:

1. I'm not 100% sure that `system_name` is not empty or `nil` in all cases, there could be corner cases
2. The title has some uniqueness restrictions, one good way to ensure `title` is unique is setting it to the `id`, which must be unique.

I don't think it's setting the `id` as title is a big deal. When a section has no title, there's no way the user can click on it to edit/delete it. That's probably the only reason why there are sections with empty titles: they were created somehow, and then they can't be removed. See the attached image:

[![image](https://user-images.githubusercontent.com/17052241/208913026-8203313a-0f25-4630-aa0d-625e6bb3f606.png)]

For theses cases, having anything to click on will be helpful enough, they can change the title later or remove the section.

Reply:
Well, there are currently to be the following validations:
```
validates :system_name, :provider, presence: true
validates :system_name, uniqueness: { :scope => [:provider_id] }, length: { maximum: 255 }
```

So I guess the system name is something that is expected to be present always (unlike the title for which the presence validation was missing).

Using system name value to fill the title is just more consistent with how title/name and system name are related for most of the models in Porta. Also, some suggestions here: https://github.com/3scale/porta/pull/3140#issuecomment-1400504580

Reply:
OK, I checked all the git history from the beginning in the system repo and it seems the `system_name` presence validation has been there the whole time. So it should be impossible to have any section in the DB without a `system_name`. I also queried `CMS::Sections.where(system_name: [nil, ''])` in SaaS to verify, and it returned 0 records. It updated the task.